jwt-bearer-auth-yesod/src/Web/Auth/Bearer/JWT/Yesod.hs

```diff
@@ -124,3 +131,22 @@ deriving via
   (ReaderT (HandlerData site site) IO)
   instance
     MonadTime (HandlerFor site)
+
+withJWKStore
+  :: MonadUnliftIO m
+  => JWTBearerAuthSettings store
+  -> (ConfiguredStore store -> m a)
+  -> m a
+withJWKStore
+  settings@JWKCacheSettings {jwkCacheTokenServerUrl, jwkCacheRefreshDelayMicros}
+  f =
+    JWKCache.withJWKCache jwkCacheRefreshDelayMicros jwkCacheTokenServerUrl
+      $ f . ConfiguredStore settings
+withJWKStore
+  settings@StaticJWKSettings {staticJWK}
+  f =
+    f (ConfiguredStore settings staticJWK)
+withJWKStore
+  settings@TokenServerSettings {tokenServerUrl}
+  f =
```

Comment:
I think this would read better with `LambdaCase`.

Also, and apologies for bringing this up now, but in our newer libraries we've been happy to drop older GHC supper enough that we can use overloaded record dot and no field selectors pattern we like generally.

Reply:
> TBH that looks about the same to me

I think the big difference is when the indentation _moves back_ after a few lines, that's jarring. I like the idea that the function definition is all indented _or_ that each expression is a single line so it never indents then comes back.

Go with how you feel, but that's why I'd consider it different/better to use `LambdaCase`.